packages/cache/src/adapter.ts

```diff
@@ -0,0 +1,9 @@
+export default class CacheAdapter {
+  static new(_cacheAdapterConstructor: typeof CacheAdapter, args: any): () => Promise<CacheAdapter> {
```

Comment:
I Commented about the errors, do not use Error class directly. Implement something like

```ts
export class NotImplementedAdapterException extends Error {
  constructor(methodName: string) {
    super(`Method ${methodName} was not implemented in your Adapter, it should be in order to fully work.`);
    this.name = NotImplementedAdapterException.name;
  }
  ```
  
  this is from the Databases package, i'm creating one from each package so they don't clash and can be properly handled, but they are pretty much copy and paste.
  
  For schemas i have this one
  ```ts
  export class SchemaAdapterNotImplementedError extends Error {
  constructor(args: { className: string; functionName: string }) {
    super(`Schema adapter did not implement ${args.functionName} in ${args.className}`);
  }
}
  ```
  
  Just that. Same as the cache, do not forget the other methods.
  
  Set will be kind of special because TTL

Reply:
Done